protocol.md

```diff
@@ -1199,7 +1199,7 @@ interface ParameterInformation {
 
 #### Goto Definition Request
 
-The goto definition request is sent from the client to the server to resolve the definition location of a symbol at a given text document position.
+The goto definition request is sent from the client to the server to resolve the definition location of a symbol at a given text document position. A definition location should span the entire body of the symbol declaration.
```

Comment:
1. `...should span the...` `span` should be `range` because that is the actual type (`Range` not `Span`, there is no span in LSP).
2. `...of the symbol declaration.` I think that `declaration` isn't the right term, shouldn't `definition` be used instead?
3. I think it's not clear what a 'body' is, maybe provide an example.

I would use something like:

> The location is the entire range of the definition, e.g. the entire range of a function starting from it's signature to the final closing code bracket.